### ISSUE TITLE
Bruteforce Method Taking Too Long

### ISSUE THREAD

**Issue by elvinguitar** [aboul3la/Sublist3r#26]
It says that subbrute is integrated in this tool, and subbrute claims to be the fastest subdomain enum tool. When sublist3r is running the bruteforce method using the subbrute, it takes really long time to finish. Sometimes, the bruteforce is running for around 3-4 hrs and it still not yet finished. Is it normal?

**Comment by aboul3la** [comment]
Yes subbrute should be much faster than this and it shouldn't take 3-4 hrs to finish enumeration. But the new subbrute wordlist is huge with 129k+ records, so it would take more time than the smaller lists.

From my own tests the whole enumeration process took around ~ 25 mins. but i was using more threads in subbrute. So my advice to you is to try the enumeration again using more threads using -t option like -t 40 or -t 50 as the following.
```
python sublist3r.py -d domain.com -b -t 50
```

Also make sure that your Internet connection is fast and reliable.

**Comment by elvinguitar** [comment]
I already tried using 50 threads but it's still taking hours to complete. The problem must be with my ISP. Anyway, thanks for answering my question.

**Comment by aboul3la** [comment]
@elvinguitar, Just to make sure did you run Sublist3r on MAC OSX? 
As I realized there was a bug in the subbrute version I used with the OSX, it doesn't work properly.
But I'm gonna push a fix for that in few days.

**Comment by elvinguitar** [comment]
Yes, im running them in OSX


Sent from Outlook for Android






On Wed, Sep 7, 2016 at 9:35 AM +1000, "Ahmed Aboul-Ela" <notifications@github.com> wrote:












@elvinguitar, Just to make sure did you run Sublist3r on MAC OSX? 

As I realized there was a bug in the subbrute version I used with the OSX, it doesn't work properly.

But I'm gonna push a fix for that in few days.

**Comment by paranymph** [comment]
I'm having the same problem with Sublist3r freezing once it invokes subbrute on OSX. I even left it running over night to be sure.

if it's any help, when i try to run subbrute.py it always fails:

```
$ python subbrute.py yahoo.com -v
('yahoo.com', False, '/Users/nymph/Sublist3r/subbrute/names.txt', '/Users/nymph/Sublist3r/subbrute/resolvers.txt', 16, False, False)
wildcard exception: ['195.221.20.10'] <class 'dns.resolver.NoNameservers'> 
Rejected nameserver - wildcard: 195.221.20.10 
Added nameserver: 205.172.19.193 
wildcard exception: ['173.241.133.189'] <class 'dns.exception.Timeout'> 
Rejected nameserver - wildcard: 173.241.133.189 
Added nameserver: 203.130.2.3 

[the program freezes here]
```

**Comment by warlock0007** [comment]
Any fix, published to solve this long time delay problem in OS X while using brute force module?
